core/server/server.go

```diff
@@ -49,10 +50,16 @@ func NewCoreConfig(cfg *rest.Config, clusterName string) CoreServerConfig {
 
 func NewCoreServer(cfg CoreServerConfig) pb.CoreServer {
 	cfgGetter := kube.NewImpersonatingConfigGetter(cfg.RestCfg, false)
+	k8s := kube.NewDefaultClientGetter(cfgGetter, cfg.clusterName)
+	ctx := context.Background()
```

Comment:
That's a context mess right now.

Reply:
Was there a reason we cannot pass it in from the caller?